httpServerIncompleteMessage is flaky

httpServerIncompleteMessage
AssertionError: Values are not equal:
error: Test failed


    [Diff] Actual / Expected


-   connection error:Connectionresetbypeer(oserror54)
+   connection closedbeforemessagecompleted


    at assertEquals (file:///Users/runner/work/***/***/test_util/std/testing/asserts.ts:271:9)
    at httpServerIncompleteMessage (file:///Users/runner/work/***/***/cli/tests/unit/http_test.ts:866:7)
    at async asyncOpSanitizer (***:runtime/js/40_testing.js:35:9)
    at async resourceSanitizer (***:runtime/js/40_testing.js:72:7)
    at async exitSanitizer (***:runtime/js/40_testing.js:99:9)
    at async applyPermissions (***:runtime/js/40_testing.js:124:9)
    at async runTest (***:runtime/js/40_testing.js:232:7)
    at async Object.runTests (***:runtime/js/40_testing.js:317:22)

failures:

	httpServerIncompleteMessage

test result: FAILED. 868 passed; 1 failed; 21 ignored; 0 measured; 0 filtered out (118626ms)

--- a/cli/tests/unit/http_test.ts
+++ b/cli/tests/unit/http_test.ts
@@ -863,37 +863,25 @@ Deno.test(
     const ev = await httpConn.nextRequest();
     const { respondWith } = ev!;
 
-    const { readable, writable } = new TransformStream<Uint8Array>();
-    const writer = writable.getWriter();
-
-    async function writeResponse() {
-      await delay(50);
-      await writer.write(
-        new TextEncoder().encode(
-          "written to the writable side of a TransformStream",
-        ),
-      );
-      await writer.close();
-    }
-
     const errors: Error[] = [];
 
-    const writePromise = writeResponse()
-      .catch((error: Error) => {
+    const readable = new ReadableStream({
+      async pull(controller) {
+        client.close();
+        await delay(100);
+        controller.enqueue(new TextEncoder().encode(
+          "written to the writable side of a TransformStream",
+        ));
+        controller.close();
+      },
+      cancel(error) {
         errors.push(error);
-      });
+      },
+    });
 
     const res = new Response(readable);
 
-    const respondPromise = respondWith(res)
-      .catch((error: Error) => errors.push(error));
-
-    client.close();
-
-    await Promise.all([
-      writePromise,
-      respondPromise,
-    ]);
+    await respondWith(res).catch((error: Error) => errors.push(error));
 
     httpConn.close();
     listener.close();

